### ISSUE TITLE
Only create a cart if at least one product was added

### ISSUE THREAD

**Issue by tobias-kuendig** [OFFLINE-GmbH/oc-mall-plugin#280]
```php
    public static function putSession($sessionId)
    {
        Cookie::queue('cart_session_id', $sessionId, 9e6);
        Session::put('cart_session_id', $sessionId);
    }

    public static function createSession(): Cart
    {
        $sessionId = Session::get('cart_session_id') ?? Cookie::get('cart_session_id') ?? str_random(100);
        self::putSession($sessionId);
        return self::orderBy('created_at', 'DESC')->firstOrCreate(['session_id' => $sessionId]);
    }

    protected static function fromSession(): ?Cart
    {
        $sessionId = Session::get('cart_session_id') ?? Cookie::get('cart_session_id');
        if (Session::get('cart_session_id')) {
            self::putSession($sessionId);
        }
        return self::where('session_id', $sessionId)->orderBy('created_at', 'DESC')->first();
    }
```

Credits to samuell on Slack

**Comment by RobertCh2** [comment]
Is this implemented? My table offline_mall_carts is getting very big with time, maybe this is the reason?

**Comment by tobias-kuendig** [comment]
No it is not implemented yet but probably the reason your table is large. You can use a query like this to get old carts and delete them:

```php
$q = \OFFLINE\Mall\Models\Cart
            ->where('customer_id', null)
            ->where('updated_at', '<', now()->subDays(3))
            ->get()
            ;
```

**Comment by RobertCh2** [comment]
The table is getting very large very quickly, like 500-600 new carts a day. Every few minutes, irregular intervals. I don't have such a traffic on my site. What could be the reason?

**Comment by tobias-kuendig** [comment]
Do you have server access logs? Could be bots indexing your site

**Comment by RobertCh2** [comment]
I have checked logs. Indeed the traffic is from bots, mainly Googlebot, bingbot and PetalBot. So I have to clean the Carts table very often.

**Comment by tobias-kuendig** [comment]
Sometimes a small change goes a long way. This is fixed in v3.4.3 (107ed121)
